src/Hungary/Mews.Fiscalizations.Hungary/Models/Invoice/ForeignCompany.cs

```diff
@@ -21,11 +21,24 @@ private ForeignCompany(Name name, SimpleAddress address, TaxpayerIdentificationN
 
         public static ITry<ForeignCompany, IEnumerable<Error>> Create(Name name, SimpleAddress address, TaxpayerIdentificationNumber taxpayerId = null)
         {
-            return Try.Aggregate(
+            var result = Try.Aggregate(
                 ObjectValidations.NotNull(name),
                 ObjectValidations.NotNull(address),
-                (n, a) => new ForeignCompany(n, a, taxpayerId)
+                (n, a) => taxpayerId.ToOption().Match(
+                    i => IsForeignTaxpayerNumber(i).ToTry(
+                        t => new ForeignCompany(n, a, i),
+                        f => new Error($"{nameof(TaxpayerIdentificationNumber)} must be a foreign (non-Hungarian) taxpayer number.").ToEnumerable()
+                    ),
+                    _ => Try.Success<ForeignCompany, IEnumerable<Error>>(new ForeignCompany(n, a))
+                )
```

Comment:
Just do 
var id = taxpayerId.ToOption().Where(i => IsForeignTaxpayerNumber(i)).ToTry(....)

And then put it into the try aggregate as a third Try, so you don't need to flatmap.